lib/modules/manager/npm/artifacts.ts

```diff
@@ -0,0 +1,77 @@
+import { logger } from '../../../logger';
+import { exec } from '../../../util/exec';
+import { ExecOptions } from '../../../util/exec/types';
+import { readLocalFile } from '../../../util/fs';
+import { UpdateArtifact, UpdateArtifactsResult } from '../types';
+
+// used to add the digest recommened by corepack
+// since the digest is generated and added by corepack itself and cannot be fecthed from any npm regsitry
+export async function updateArtifacts({
+  packageFileName,
+  config,
+  updatedDeps,
+  newPackageFileContent: existingPackageFileContent,
+}: UpdateArtifact): Promise<UpdateArtifactsResult[] | null> {
+  logger.debug(`npm.updateArtifacts(${packageFileName})`);
+  const packageManagerUpdate = updatedDeps.find(
+    (dep) => dep.depType === 'packageManager',
+  );
+
+  if (!packageManagerUpdate) {
+    logger.debug('No packageManager updates - returning null');
+    return null;
+  }
+
+  // this adds a lockfile if not present
+  // but I have not added that to the list of changed files
+  // will that suffice, or should I do a proper cleanup so that it does not create problems with other updates
+  let cmd = `corepack use `;
```

Comment:
Running this command creates a lockfile is one is not present. Should I check & delete it in case one is created or should I let it be as we do not return the file and it won't be staged & committed.

Reply:
we should ignore this case as it shouldn't happen. the file is either in git ignore or lock file is disabled via npmrc.